packages/at_client/lib/src/service/notification_service.dart

```diff
@@ -101,6 +101,31 @@ abstract class NotificationService {
 
   /// Returns the status of the given notificationId
   Future<NotificationResult> getStatus(String notificationId);
+
+  /// Returns the [AtNotification] of the given notificationId
+  ///
+  /// Usage:
+  ///
+  ///  To fetch a notification with id: abc-123
+  /// ```dart
+  /// var notification = NotificationServiceImpl(atClient!);
+  /// var atNotification = await notification.fetch('abc-123');
+  /// ```
+  /// Returns [NotificationStatus.delivered] when notification exist in the key-store
+  /// ```
+  ///  AtNotification{id: abc-123, key: @bob:phone.wavi@alice,
+  ///  from: @alice, to: @bob, epochMillis: 1666268427030, value: "+1 887 887 7876",
+  ///  operation: OperationType.update, status: NotificationStatus.delivered}
+  /// ```
+  ///
+  /// Returns [NotificationStatus.expired] when
+  /// * Notification id does not exist
+  /// * Notification is expired / delete
+  /// ```
+  ///  AtNotification{id: abc-123, status: NotificationStatus.expired}
+  /// ```
+  /// Throws [AtClientException] when server is not reachable or server timeout's to respond
```

Comment:
What does it return for nonexistent notification?

Reply:
For a non-existent notification also, we return `AtNotification{id: abc-123, status: NotificationStatus.expired}`.

Currently, we do not have a way to differentiate between if the notification is expired and deleted or if the notification does not exist. So, returning the same response for both.

Updated the same in the dart docs too.

Reply:
I think this is OK